test/functional/syscalls-e2e-specs.js

```diff
@@ -29,4 +31,29 @@ describe('System calls', () => {
   it('shell should execute command in adb shell ', async () => {
     (await adb.shell(['getprop', 'ro.build.version.sdk'])).should.equal(apiLevel);
   });
+  it('getConnectedEmulators should get all connected emulators', async () => {
+    (await adb.getConnectedEmulators()).length.should.be.above(0);
+  });
+  it('getRunningAVD should get all connected avd', async () => {
+    await adb.getRunningAVD(avdName).should.not.be.null;
+  });
+  it('getRunningAVDWithRetry should get all connected avds', async () => {
+    await adb.getRunningAVDWithRetry(avdName).should.not.be.null;
+  });
+  // Skipping for now. Will unskip depending on how it behaves on CI
+  it.skip('launchAVD should get all connected avds', async function () {
+    this.timeout(240000);
+    let proc = await adb.launchAVD(avdName);
+    (await adb.getConnectedEmulators()).length.should.be.above(0);
+    proc.kill();
+  });
+  it('waitForDevice should get all connected avds', async function () {
+    this.timeout(60000);
```

Comment:
launching an emu takes a long time :-/

ultimately I don't think it should be Appium's responsibility to do this and we should deprecate this behavior, but since we don't want breaking changes here, we need to include it in this rewrite.